### DISCUSSION TITLE
Ray tracing coverage map and path figure are inconsistent

### DISCUSSION THREAD

**Discussion by Mio-TW** [NVlabs/sionna#477] [Q&A] [answered]
Dear Sionna Team,

I loaded the default scene "munich" in Sionna and created a TX and multiple RX points.
Here is my code:

```py
scene = load_scene(sionna.rt.scene.munich) 
resolution = [480,320]

# Configure antenna array for all transmitters
scene.tx_array = PlanarArray(num_rows=1,
                             num_cols=1,
                             vertical_spacing=0.5,
                             horizontal_spacing=0.5,
                             pattern="tr38901",
                             polarization="V")

# Configure antenna array for all receivers
scene.rx_array = PlanarArray(num_rows=1,
                             num_cols=1,
                             vertical_spacing=0.5,
                             horizontal_spacing=0.5,
                             pattern="dipole",
                             polarization="cross")
                             
# Create transmitter
tx = Transmitter(name="tx",
                 position=[8.5,21,27])
scene.add(tx)

### RX1 ###
rx = Receiver(name="rx",
              position=[45,90,1.5],
              orientation=[0,0,0])
scene.add(rx)
tx.look_at(rx) 

### RX2 ###
rx_2 = Receiver(name="rx_2",
              position=[20,130,1.5],
              orientation=[0,0,0])
scene.add(rx_2)
tx.look_at(rx_2)

### RX3 ###
rx_3 = Receiver(name="rx_3",
              position=[90,20,1.5],
              orientation=[0,0,0])
scene.add(rx_3)
tx.look_at(rx_3)

### RX4 ###
rx_4 = Receiver(name="rx_4",
              position=[200,200,1.5],
              orientation=[0,0,0])
scene.add(rx_4)
tx.look_at(rx_4)


# Set carrier frequency
scene.frequency = 3.5e9 # in Hz

# Compute propagation paths
paths = scene.compute_paths(max_depth=5, reflection=True, diffraction=False, scattering=False)  

# Create new camera with different configuration
cam_TxRx = Camera("cam_TxRx", position=[-400,80,300], look_at=[20,100,50])
scene.add(cam_TxRx)

# Visualize paths in the 3D preview
scene.render("cam_TxRx", paths=paths, resolution=resolution, num_samples=512)
```
![PXL_20240612_025624011~2](https://github.com/NVlabs/sionna/assets/151720216/c4806b20-8891-4c68-8eae-4c2f80a1435a)

```py
cm = scene.coverage_map(max_depth=5,
                        reflection=True,
                        diffraction=False, 
                        scattering=False,
                        cm_cell_size=(5., 5.),
                        combining_vec=None,
                        precoding_vec=None,
                        num_samples=int(1e6)) 

# Create new camera
tx_pos = scene.transmitters["tx"].position.numpy()
bird_pos = tx_pos.copy()
bird_pos[-1] = 1000 # Set height of coverage map to 1000m above tx
bird_pos[-2]-= 0.01 # Slightly move the camera for correct orientation

# Create new camera
bird_cam = Camera("birds_view", position=bird_pos, look_at=tx_pos)
scene.add(bird_cam)

scene.render("birds_view", coverage_map=cm, resolution=resolution, num_samples=512); 
```
![PXL_20240612_025525751~2](https://github.com/NVlabs/sionna/assets/151720216/e551a734-0564-4460-a58e-71b31162ebd7)

I've visualized the path and coverage map as shown above.
However, you can see from the coverage map that RX4 can be covered.
But in the path figure, RX4 does not receive any path.
What is the problem causing this inconsistency?

Thank you so much for help.

**Comment by jeertmans** [comment]
Hello, Sionna authors correct me if I am wrong, but I think this is because the first plot uses an approximate point to point ray tracing (approximate in the sens that not all path candidates are tried, since you probably use the Fibonacci method), and the second plot uses Ray Launching, another approximate Ray Tracing technique.

Both methods have their pros and cons, and I guess this is what you observe here: RX4 is not reachable by the paths selected by the Fibonacci method. Maybe you can try increasing the number of paths in the first method?

**Reply by jhoydis** [reply, marked as answer]
Hi @Mio-TW,

The issue is that a coverage map consist of tiles. If a tile receives some energy, this does not mean that all points within this tile can actually be reached; especially since you are considering only specular paths. What you observe is probably not related to the path tracing method as mentioned by @jeertmans (although it could be).

**Reply by jeertmans** [reply]
Thanks for the clarification @jhoydis! I guess you (@Mio-TW) can try reducing the tile size and see if the coverage of RX4 changes with that.

**Reply by jeertmans** [reply]
@Mio-TW for the update, @jhoydis's explanation seems to be the right one: reducing the size of the tiles (from 5m x 5m to 1m x 1m) seems to show that RX4 is not in an area covered by TX.

![image](https://github.com/NVlabs/sionna/assets/27275099/4faf0a63-322d-4ba6-afc7-c9d40568cd17)

**Reply by Mio-TW** [reply]
@jhoydis @jeertmans, thank you both for your assistance. 
@jeertmans, I also tried to reduce the tiles size and got the same result as you, I truly appreciate it :) Thank you!
